go/parquet/pqarrow/column_readers.go

```diff
@@ -443,14 +454,16 @@ func chunksToSingle(chunked *arrow.Chunked) (arrow.ArrayData, error) {
 	case 0:
 		return array.NewData(chunked.DataType(), 0, []*memory.Buffer{nil, nil}, nil, 0, 0), nil
 	case 1:
-		return chunked.Chunk(0).Data(), nil
+		data := chunked.Chunk(0).Data()
+		data.Retain() // we pass control to the caller
```

Comment:
we want the caller to have the full control of the data (so, releasing the array after the `chunksToSingle` should have no effect on the returned data.
For 0 len it was already true (we constructed a new data), so this retain comes in naturally